src/main/java/it/systems/paymentreviser/repository/MemoryRepository.java

```diff
@@ -0,0 +1,14 @@
+package it.systems.paymentreviser.repository;
+
+import it.systems.paymentreviser.entity.Case;
+
+import java.util.List;
+
+public interface MemoryRepository {
```

Comment:
This interface could potentially be implemented with both an in-memory repository and a database. Therefore, I don't think it should have "Memory" in its name. The interface could be `CaseRepository`, while the in-memory implementation could be `CaseMemoryRepository` or something like that.